controllers/v1/solutionsController.js

```diff
@@ -1558,5 +1548,161 @@ module.exports = class Solutions extends Abstract {
 
         })
     }
+
+    /**
+    * @api {post} /assessment/api/v1/solutions/addEntities/:solutionId Add entity to solution
+    * @apiVersion 1.0.0
+    * @apiName Add entity to solution
+    * @apiGroup Solutions
+    * @apiParamExample {json} Request-Body:
+    * {
+    *	    "entities": ["5beaa888af0065f0e0a10515","5beaa888af0065f0e0a10516"]
+    * }
+    * @apiHeader {String} X-authenticated-user-token Authenticity token
+    * @apiSampleRequest /assessment/api/v1/solutions/addEntities/5f64601df5f6e432fe0f0575
+    * @apiUse successBody
+    * @apiUse errorBody
+    * @apiParamExample {json} Response:
+    * {
+    *   "message" : "Entities updated successfully."
+    * }
+    */
+
+     /**
+   * Add entity to solution.
+   * @method
+   * @name addEntities
+   * @param {Object} req - requested data.
+   * @param {String} req.params._id - solution id.
+   * @returns {JSON} consists message of successfully mapped entities
+   */
+
+  async addEntities(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let solutionData = await solutionsHelper.addEntityToSolution(
+          req.params._id,
+          req.body.entities
+        );
+
+        return resolve(solutionData);
+
+      } catch (error) {
+        return reject({
+          status: error.status || httpStatusCode.internal_server_error.status,
+          message: error.message || httpStatusCode.internal_server_error.message,
+          errorObject: error
+        });
+      }
+    });
+  }
+
+
+    /**
+    * @api {post} /assessment/api/v1/solutions/list list of solutions
+    * @apiVersion 1.0.0
+    * @apiName List solutions
+    * @apiGroup Solutions
+    * @apiParamExample {json} Request-Body:
+    * {
+    *	    "solutionIds": ["EF-DCPCR-2018-001"]
+    * }
+    * @apiHeader {String} X-authenticated-user-token Authenticity token
```

Comment:
@aman-tunerlabs - I believe this has to be accessible via internal access token only and the required projection fields to be sent in the request body as well